core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -6524,6 +6524,15 @@ public static List reverse(List list) {
     return list;
   }
 
+  /** Support the ARRAY_SLICE function. */
+  public static List arraySlice(List list, int start, int length) {
+    // return empty list if start/length are out of range of the array
```

Comment:
How about "... if start + length exceeds the boundary of the array"?

Reply:
Would return empty array，related to ： https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArraySlice.java